### ISSUE TITLE
expending lint [`blocks_in_if_conditions`] to check not only `if`'s conditions, but also `match`'s as welll

### ISSUE THREAD

**Issue by J-ZhengLi** [rust-lang/rust-clippy#11814]
### What it does

Currently only the blocks in if conditions are warned, however, if readability was a concern then those blocks after  `match` keyword should be treated the same right?

[playground example](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=9b7b4a803ef2789325ec59e5796353a9)

### Advantage

- Readability
- Consistancy with [`blocks_in_if_conditions`]

### Drawbacks

~Could breaks auto-fix capability.~

~Check the second example, which a block was used in a `while` condition, however, if we move the block up and bind it as suggested, the code will not work, as x does not changes anymore causing an infinite loop.~

~But maybe we can skip such senario.~

### Example

```rust
fn match_block() {
    let _x: i32 = match {
        let opt: Option<u8> = Some(1);
        opt
    } {
        Some(num) => num.into(),
        None => -1
    };
}
```

Could be written as:

```rust
fn match_block() {
    let opt: Option<u8> = Some(1);
    let _x: i32 = match opt {
        Some(num) => num.into(),
        None => -1
    };
}
```

**Comment by Jarcho** [comment]
The change on `while` is not the same since it needs to be evaluated every iteration.

**Comment by J-ZhengLi** [comment]
> The change on `while` is not the same since it needs to be evaluated every iteration.

Good point! I've removed `while` from description
